elixir/apps/domain/lib/domain/auth/adapters/mock/jobs/sync_directory.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Domain.Auth.Adapters.Mock.Jobs.SyncDirectory do
+  use Domain.Jobs.Job,
+    otp_app: :domain,
+    every: :timer.minutes(1),
+    executor: Domain.Jobs.Executors.Concurrent
+
+  alias Domain.Auth.Adapter.OpenIDConnect.DirectorySync
+  require Logger
+  require OpenTelemetry.Tracer
+
+  @task_supervisor __MODULE__.TaskSupervisor
+
+  @impl true
+  def state(_config) do
+    {:ok, pid} = Task.Supervisor.start_link(name: @task_supervisor)
+    {:ok, %{task_supervisor: pid}}
+  end
+
+  @impl true
+  def execute(%{task_supervisor: pid}) do
+    DirectorySync.sync_providers(__MODULE__, :mock, pid)
+  end
+
+  def gather_provider_data(provider, _task_supervisor_pid) do
+    num_groups = provider.adapter_config["num_groups"]
+    num_actors = provider.adapter_config["num_actors"]
+    max_actors_per_group = provider.adapter_config["max_actors_per_group"]
+
+    identities_attrs =
+      1..num_actors
+      |> Enum.map(fn i ->
+        first_name = Domain.NameGenerator.generate_first_name()
+        last_name = Domain.NameGenerator.generate_last_name()
+
+        %{
+          "provider_identifier" => "U:#{i}",
+          "provider_state" => %{
+            "userinfo" => %{
+              "email" => "#{String.downcase(first_name)}@example.com"
```

Comment:
Won't this likely lead to duplicate email addresses?

Reply:
So the `provider_identifier` is deterministic and consistent across runs - what this exercises is how well we are able to upsert email and other fields related to users and groups from the provider, assuming the list of `provider_identifier`s returned for each is consistent.

Notably, this adapter _does not_ test how we handle actual diffs very well, which could be a good addition to add later. The main goal here was just to have some data in dev to test with.

Reply:
Gotcha.  I suppose this is kind of a "worst case" scenario potentially.

It would be nice to have the ability to set how "aggressive" the diffs are on the sync, but that can be added later if we really need it.